packages/jsx-email/src/components/conditional.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { Suspense } from 'react';
+
+import { jsxToString, useData } from '../render/jsx-to-string';
+import type { JsxEmailComponent } from '../types';
+
+export interface ConditionalProps {
+  children?: React.ReactNode;
+  expression?: string;
+  mso?: boolean;
+}
+
+const notMso = (html: string) => `<!--[if !mso]><!-->${html}<!--<![endif]-->`;
+
+const comment = (expression: string, html: string) => `<!--[if ${expression}]>${html}<![endif]-->`;
+
+const Renderer = (props: ConditionalProps) => {
+  const { children, mso } = props;
+  let { expression } = props;
+  const html = useData(props, () => jsxToString(<>{children}</>));
+  let innerHtml = '';
+
+  if (mso === false) innerHtml = notMso(html);
+  else if (mso === true && !expression) expression = 'mso';
+  if (expression) innerHtml = comment(expression, html);
+
+  // @ts-ignore
```

Comment:
That change will remove this ts-ignore usage
and this will now be

```tsx
<Jsx-email-cond dangerouslySetInnerHTML={{ __html: innerHtml }} />
```

ultimately it renders a html comment in the email template